src/addons/addon.cpp

```diff
@@ -209,41 +210,51 @@ QList<ConditionCallback> s_conditionCallbacks{
        // dynamic condition
        return true;
      },
-     [](QObject* parent, const QJsonValue& value) -> AddonConditionWatcher* {
+     [](Addon* addon, const QJsonValue& value) -> AddonConditionWatcher* {
        QStringList locales;
        for (const QJsonValue& v : value.toArray()) {
          locales.append(v.toString().toLower());
        }
 
-       return AddonConditionWatcherLocales::maybeCreate(parent, locales);
+       return AddonConditionWatcherLocales::maybeCreate(addon, locales);
      }},
 
     {"trigger_time",
      [](const QJsonValue&) -> bool {
        // dynamic condition
        return true;
      },
-     [](QObject* parent, const QJsonValue& value) -> AddonConditionWatcher* {
+     [](Addon* addon, const QJsonValue& value) -> AddonConditionWatcher* {
        return AddonConditionWatcherTriggerTimeSecs::maybeCreate(
-           parent, value.toInteger());
+           addon, value.toInteger());
      }},
 
     {"start_time",
      [](const QJsonValue&) -> bool {
        // dynamic condition
        return true;
      },
-     [](QObject* parent, const QJsonValue& value) -> AddonConditionWatcher* {
-       return new AddonConditionWatcherTimeStart(parent, value.toInteger());
+     [](Addon* addon, const QJsonValue& value) -> AddonConditionWatcher* {
+       return new AddonConditionWatcherTimeStart(addon, value.toInteger());
      }},
 
     {"end_time",
      [](const QJsonValue&) -> bool {
        // dynamic condition
        return true;
      },
-     [](QObject* parent, const QJsonValue& value) -> AddonConditionWatcher* {
-       return new AddonConditionWatcherTimeEnd(parent, value.toInteger());
+     [](Addon* addon, const QJsonValue& value) -> AddonConditionWatcher* {
+       return new AddonConditionWatcherTimeEnd(addon, value.toInteger());
+     }},
+
+    {"javascript",
+     [](const QJsonValue&) -> bool {
+       // dynamic condition
+       return true;
+     },
+     [](Addon* addon, const QJsonValue& value) -> AddonConditionWatcher* {
+       return AddonConditionWatcherJavascript::maybeCreate(addon,
+                                                           value.toString());
```

Comment:
Question: can't we substitute almost all of the current condition watchers with JS condition watchers?

Reply:
in theory yes, but it would make all more complex to follow and implement.